### PR TITLE
Listeners will not be reset on reconnection to the message broker

### PR DESCRIPTION
**Pull request by krlvi**
Hi, why are the listeners reset after a reconnection and re-publishing of the connections?
In `def handle_cast({:subscribe_connection, listener}, {host, connection, listeners})` new listeners are added to the state. 
However the callback `def handle_info(:connect, {host, _, listeners})` sets it back to `[]`

### CODE REVIEW

**Comment by ono** [comment]
@krlvi the connection needs to send the notification only once. Workers don't need to receive the message twice.

`subscribe_conneciton/2` handles two situation.

1. when the connection has not been established
2. when the connection has been established

for the later, it will send the message immediately to the caller and the message shouldn't be sent to other listeners.

Are there any reasons you want to keep listeners?

**Comment by krlvi** [comment]
@ono I was studying the code and thinking about how would the application handle a loss of rabbit connection. In which case `handle_info({:DOWN` could try to handle_info(:connect with the same subscribers. I am not sure if this is a desired behaviour however.

**Comment by ono** [comment]
I see. Let me try to explain how it works on the reconnection.

1. When the rabbitmq connection is lost, TaskBunny Connection process [kill itself](https://github.com/shinyscorpion/task_bunny/blob/master/lib/task_bunny/connection.ex#L200).
2. The supervisor detects the stop so it will respawn Connection process. Since [its supervising strategy is `one_for_all`](https://github.com/shinyscorpion/task_bunny/blob/master/lib/task_bunny/supervisor.ex#L50), the supervisor will also restart all workers through [WorkerSupervisor](https://github.com/shinyscorpion/task_bunny/blob/master/lib/task_bunny/worker_supervisor.ex).
3. Restarted workers will subscribe for a new connection. The new connection process will notify them once the connection is established.

So basically when the connection is renewed, there will be new set of processes. If you are interested in looking more details, I would suggest to start TaskBunny application in iex. Open Observer and check the processes. Then stop RabbitMQ and start it again. You will see how it handles the reconnection.

**Comment by krlvi** [comment]
@ono Thanks for the explanation, makes sense, and this seems to be the elixir way of doing this. I am closing this PR
